cli/bazelisk/bazelisk.go

```diff
@@ -144,12 +156,18 @@ func redirectStdio(w io.Writer, stdio **os.File) (close func(), err error) {
 		closePipe = c
 		f = pw
 	}
-	*stdio = f
+	original := make([]*os.File, len(stdio))
+	for i := range stdio {
+		original[i] = *stdio[i]
+		*stdio[i] = f
```

Comment:
not sure I understand how this fixes the race condition - we're still redirecting both stdout and stderr to the same writer `w`, but `Write` can be called concurrently on both stdout and stderr, no?

Reply:
We're redirecting them to a pipe writer, though, which is a File, and so the write calls are syscalls, which are threadsafe.

Reply:
ah, got it